File: aoc2018/python-solutions/day05.py
def solve(inp: str) -> int:
    res = []
    for y in inp:
        if res == [] or y.swapcase() != res[-1]:
            res.append(y)
        else:  # Aa aA
            res.pop()
    return len(res)


with open("../inputs/day05.txt") as f:
    inp = f.readlines()[0].strip()
    print(solve(inp))
    print(
        min(
            map(
                solve,
                [
                    "".join(x for x in inp if x.lower() != u)
                    for u in set([x for x in inp if x.islower()])
                ],
            )
        )
    )

File: aoc2019/c_solutions/day05.c
#include "intcode.h"

int main() {
  sz n, *inp = read_ints("../inputs/day05.txt", &n);
  CHECK(ic_run_util_fire(inp,n,1), 9654885);
  CHECK(ic_run_util_fire(inp,n,5), 7079459);
  free(inp);
}
